sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1974,7 +1975,10 @@ case class ElementAt(left: Expression, right: Expression) extends GetMapValueUti
     }
   }
 
-  override def nullable: Boolean = true
+  override def nullable: Boolean = left.dataType match {
+    case _: ArrayType => computeNullabilityFromArray(left, right)
+    case _: MapType => computeNullabilityFromMap
```

Comment:
If `Map` is out of the scope, shall we update the title and use the following code path instead?
- Remove `extends GetMapValueUtil`.
- Remove `computeNullabilityFromMap` by the following.
 ```scala
case _: MapType => true
```

Reply:
fixed.